includes/Core/Modules/Modules.php

```diff
@@ -244,6 +245,41 @@ function ( Module $module ) {
 			}
 		);
 
+		add_action(
+			'googlesitekit_authorize_user',
+			function( $token_response ) {
+				if ( empty( $token_response['analytics_configuration'] ) ) {
+					return;
+				}
+
+				if ( ! $this->is_module_active( Analytics::MODULE_SLUG ) ) {
+					$this->activate_module( Analytics::MODULE_SLUG );
+				}
+
+				try {
+					$analytics = $this->get_module( Analytics::MODULE_SLUG );
+					$analytics->handle_token_response_data( $token_response );
+				} catch ( Exception $e ) {
+					return;
+				}
```

Comment:
Nit-pick, but I think this block should be moved to the bottom of this method. The part of migrating the scope is directly tied to the module not having been active before, it's basically "cleanup" around that.

The call here however just invokes the actual functionality to handle the response data, which would otherwise happen via the hook in the `Analytics` class on priority `10`, so it makes sense for this to happen last.

Technically, we could even just call the `Analytics::register` method here and then `handle_token_response_data` could remain private/protected, but I agree with the approach of calling it directly here since it's easier to follow and comes without the other registration side effects.

Reply:
Updated.